apps/libsignpost/port_signpost_tock.c

```diff
@@ -0,0 +1,140 @@
+#include <stdint.h>
+#include <string.h>
+#include "tock.h"
+#include "port_signpost.h"
+#include "i2c_master_slave.h"
+#include "gpio.h"
+#include "led.h"
+#include "timer.h"
+
+/* This is a port interface for the signpost API and networking stack!
+The goal is to export the minimal interface such that platforms
+can integrate with the signpost.
+
+From a high level you will need to provide:
+    - Blocking I2C write function (currently at least 255 bytes at a time)
+    - Asynchronous I2C slave function (listen on an address and send
+                                        back written bytes asynchronously)
+    - Set and clear of the MOD_OUT pin
+    - Asynchronous callback for falling edge of the Mod-in pin
+    - Currently the signpost libraries assume printf functionality
+        - This is assumed to be provided through <stdio.h>
+
+To provide this functionality please implement the following interface
+for each platform*/
```

Comment:
This is a copy of the text in the header, except it's missing the "way to wait" bit. I think these should be the same (or actually I think it's probably better/fine to just have the exposition in the header file)